src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php

```diff
@@ -98,7 +99,7 @@ public function onControllerArguments(ControllerArgumentsEvent $event)
         $r = new \ReflectionFunction(\Closure::fromCallable($event->getController()));
         $r = $r->getParameters()[$k] ?? null;
 
-        if ($r && (!$r->hasType() || FlattenException::class === $r->getType()->getName())) {
+        if ($r && (!$r->hasType() || FlattenException::class === $r->getType()->getName() || LegacyFlattenException::class === $r->getType()->getName())) {
```

Comment:
`if ($r && (!$r->hasType() || \in_array($r->getType()->getName(), [FlattenException::class, LegacyFlattenException::class, true])) {`

Reply:
Addressed. Thank you.